datafusion/core/src/physical_planner.rs

```diff
@@ -777,6 +777,18 @@ impl DefaultPhysicalPlanner {
                     .map(|s| (s.to_string(), arrow_schema::DataType::Null))
                     .collect::<Vec<_>>();
 
+                let keep_partition_by_columns = source_option_tuples
+                    .get("format.keep_partition_by_columns")
+                    .map(|v| v.trim() == "true")
+                    .unwrap_or(false);
+
+                let mut updated_source_options_tuples = HashMap::new();
+                for (k, v) in source_option_tuples {
+                    if k != "format.keep_partition_by_columns" {
+                        updated_source_options_tuples.insert(k.clone(), v.clone());
+                    }
+                }
+
```

Comment:
I am looking for comments and suggestion on how to improve this. Conceptually this new flag is not a format option, because it can apply regardless of the format. I was thinking about maybe creating an `enum` for these new options potentially named `CopyOptions` and then partition `source_option_tuples` by values present in the enum.

Reply:
perhaps @metesynnada or @devinjdangelo  has some suggestions or ideas?

Reply:
I agree this isn't really a "format option" and it is unfortunate that "format." was added to the option. 

I suggest we update the code here https://github.com/apache/datafusion/blob/31daf25576a4bc14851e2ee3f5d3a88eba93c62a/datafusion/sql/src/statement.rs#L884-L900 to not add the "format." to this option. Perhaps this option could have a different prefix like "hive.keep_partition_by_columns".

Reply:
I segregated `format.` and `hive.` options into to different `HashMap`s, instead of having to partition the `options` HashMap later in the code.

Reply:
Reverted this approach and continued with a single HashMap with keys having different prefixes `hive.` and `format.`, then skipping the `hive.` when setting the format options.